Mage/src/main/java/mage/game/ZonesHandler.java

```diff
@@ -438,7 +438,7 @@ private static boolean maybeRemoveFromSourceZone(ZoneChangeInfo info, Game game,
             if (event.getToZone() == Zone.BATTLEFIELD && event.getTarget() != null) {
                 event.getTarget().updateZoneChangeCounter(game, event);
```

Comment:
btw for that `TODO: wtf` comment in PermanentCard, this is where that would be called.

Reply:
Permanent’s zcc is static and can’t changed (only synced on first usage/init/put). If it changed somehow then it can be an error or wrong code usage.

Reply:
Yes, but when the permanent is put onto the battlefield, its card's zcc gets updated, and the permanent stores that updated zcc.

I don't think a permanent's zcc should move after that.